contracts/peg/NonCustodialPSM.sol

```diff
@@ -0,0 +1,456 @@
+        redeemPaused = true;
+        emit RedemptionsPaused(msg.sender);
+    }
+
+    /// @notice set secondary pausable methods to unpaused
+    function unpauseRedeem() external isGovernorOrGuardianOrAdmin {
+        redeemPaused = false;
+        emit RedemptionsUnpaused(msg.sender);
+    }
+
+    /// @notice set secondary pausable methods to paused
+    function pauseMint() external isGovernorOrGuardianOrAdmin {
+        mintPaused = true;
+        emit MintingPaused(msg.sender);
+    }
+
+    /// @notice set secondary pausable methods to unpaused
+    function unpauseMint() external isGovernorOrGuardianOrAdmin {
+        mintPaused = false;
+        emit MintingUnpaused(msg.sender);
+    }
+
+    /// @notice set the mint fee vs oracle price in basis point terms
+    function setMintFee(uint256 newMintFeeBasisPoints)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setMintFee(newMintFeeBasisPoints);
+    }
+
+    /// @notice set the redemption fee vs oracle price in basis point terms
+    function setRedeemFee(uint256 newRedeemFeeBasisPoints)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setRedeemFee(newRedeemFeeBasisPoints);
+    }
+
+    /// @notice set the target for sending all PCV
+    /// @param newTarget new PCV Deposit target for this PSM
+    function setPCVDeposit(IPCVDeposit newTarget)
+        external
+        override
+        onlyGovernorOrAdmin
+    {
+        _setPCVDeposit(newTarget);
+    }
+
+    /// @notice withdraw ERC20 from the contract
+    /// @param token address of the ERC20 to send
+    /// @param to address destination of the ERC20
+    /// @param amount quantity of ERC20 to send
+    function withdrawERC20(
+        address token,
+        address to,
+        uint256 amount
+    ) external onlyGovernorOrAdmin {
+        IERC20(token).safeTransfer(to, amount);
+        emit WithdrawERC20(msg.sender, token, to, amount);
+    }
+
+    /// @notice set the mint fee vs oracle price in basis point terms
+    function _setMintFee(uint256 newMintFeeBasisPoints) internal {
+        require(
+            newMintFeeBasisPoints <= MAX_FEE,
+            "PegStabilityModule: Mint fee exceeds max fee"
+        );
+        uint256 _oldMintFee = mintFeeBasisPoints;
+        mintFeeBasisPoints = newMintFeeBasisPoints;
+
+        emit MintFeeUpdate(_oldMintFee, newMintFeeBasisPoints);
+    }
+
+    /// @notice internal helper function to set the redemption fee
+    function _setRedeemFee(uint256 newRedeemFeeBasisPoints) internal {
+        require(
+            newRedeemFeeBasisPoints <= MAX_FEE,
+            "PegStabilityModule: Redeem fee exceeds max fee"
+        );
+        uint256 _oldRedeemFee = redeemFeeBasisPoints;
+        redeemFeeBasisPoints = newRedeemFeeBasisPoints;
+
+        emit RedeemFeeUpdate(_oldRedeemFee, newRedeemFeeBasisPoints);
+    }
+
+    /// @notice helper function to set the surplus target
```

Comment:
"surplus target" doesn't make sense here anymore right? this is just the pcv deposit